src/module.ml

```diff
@@ -130,21 +130,108 @@ module Visibility = struct
   let is_private t = not (is_public t)
 end
 
+module Obj_dir = struct
+  type t =
+    { dir: Path.t (** The main directory *)
```

Comment:
"main directory" is a bit confusing. I was thinking something like `source_dir` but it's not making sense for libraries that have more than 1 source dir (`include_subdirs` stanza). Perhaps `source_root` or `stanza_dir`?